src/foam/blob/RestBlobService.java

```
@@ -0,0 +1,164 @@
+    OutputStream os = null;
+    InputStream is = null;
+    IdentifiedBlob result = null;
+    try {
+      URL url = new URL(address_);
+      connection = (HttpURLConnection) url.openConnection();
+      
+      //configure HttpURLConnection
+      connection.setConnectTimeout(5 * 1000);
+      connection.setReadTimeout(5 * 1000);
+      connection.setDoOutput(true);
+      connection.setUseCaches(false);
+
+      //set request method
+      connection.setRequestMethod("PUT");
+
+      //configure http header
+      connection.setRequestProperty("Accept", "*/*");
+      connection.setRequestProperty("Connection", "keep-alive");
+      connection.setRequestProperty("Content-Type", "application/octet-stream");
+
+      //output blob into connection
+      long chunk = 0;
+      long size = blob.getSize();
+      long chunks = (long) Math.ceil((double) size / (double) BUFFER_SIZE);
+      Buffer buffer = new Buffer(BUFFER_SIZE, ByteBuffer.allocate(BUFFER_SIZE));
+      os = connection.getOutputStream();
+
+      while ( chunk < chunks ) {
+        buffer = blob.read(buffer, chunkOffset(chunk));
+        byte[] buf = buffer.getData().array();
+        os.write(buf, 0, (int) buffer.getLength());
+        buffer.getData().clear();
+        chunk++;
+      }
+
+      os.flush();
+
+      if ( connection.getResponseCode() == HttpURLConnection.HTTP_OK ) {
+        is = connection.getInputStream();
+        BufferedReader reader = new BufferedReader(new InputStreamReader(is));
+        String json = "";
+        String line = null;
+        while ( (line = reader.readLine()) != null ) {
+          json += line;
+        }
+
+        result = (IdentifiedBlob) (new foam.lib.json.JSONParser()).parseString(json, IdentifiedBlob.class);
+        result.setX(getX());
+      } else {
+        throw new RuntimeException("upload fail");
+      }
+    } catch ( MalformedURLException e ) {
+      e.printStackTrace();
+    } catch ( IOException e ) {
+      e.printStackTrace();
```

Might want to re-throw the exception as well, or re-throw it in a RuntimeException.